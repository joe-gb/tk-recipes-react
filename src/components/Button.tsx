import styled from "styled-components";
import React from "react";

export const Button = styled.button`
  && {
    display: block;
    text-decoration: none;
    width: 100%;
    margin: 20px auto;
    cursor: pointer;
    font-size: 16px;
    font-weight: 500;
    line-height: 18px;
    user-select: none;
    color: #fff;
    border-radius: 8px;
    background: #003097;
    border: 1px solid #003097;
    padding: 19px;
    justify-content: center;
    align-items: center;
    text-align: center;
    transition: all 0.175s ease-in-out;

    &:hover {
      background-color: #144fcc;
      border: 1px solid #144fcc;
    }
  }
`;

export const NeutralButton = styled(Button)`
  && {
    background: transparent;
    border-color: #9eabbe;
    color: #144fcc;

    &:hover {
      background-color: transparent;
      border: 1px solid #144fcc;
    }
  }
`;

export const WarningButton = styled(Button)`
  && {
    background: transparent;
    border-color: #003097;
    color: #003097;
  }
`;

const SvgCrossButton = (props: React.ButtonHTMLAttributes<any>) => (
  <button {...props}>
    <svg data-testid="svg-cross" viewBox="0 0 24 24" width="1em" height="1em">
      <path
        fillRule="evenodd"
        d="M13.414 12l4.293 4.293a1 1 0 01-1.414 1.414L12 13.414l-4.293 4.293a1 1 0 11-1.414-1.414L10.586 12 6.293 7.707a1 1 0 011.414-1.414L12 10.586l4.293-4.293a1 1 0 011.414 1.414L13.414 12z"
      ></path>
    </svg>
  </button>
);

export const CrossButton = styled(SvgCrossButton)`
  && {
    
    z-index: 1;
    margin: 5px;
    background: white;
    border: 1px solid #C9D3DF;
    box-sizing: border-box;
    box-shadow: 0px 1px 2px rgb(0 0 0 / 8%);
    border-radius: 4px;
    padding: 4px;
    height: 32px;
    width: 32px;
    &:hover {
      cursor: pointer;
      border: 1px solid #9EABBE;
      box-shadow: 0px 0px 0px 1px rgb(50 50 93 / 5%), 0px 2px 6px rgb(50 50 93 / 8%);
    }
    & svg {
      fill: #4C5566;
      color: #4C5566;
      vertical-align: middle;
      font-size: 24px;
      & path {
        fill: #4C5566;
      }

    }
`;
