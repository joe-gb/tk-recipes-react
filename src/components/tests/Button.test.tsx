import React from "react";
import { render } from "@testing-library/react";
import { Button, WarningButton, NeutralButton, CrossButton } from "../Button";
import { within } from "@testing-library/dom";

test("Primary button snapshot test", () => {
  let { container } = render(<Button>Click me</Button>);
  expect(container).toMatchSnapshot();
  expect(container.textContent).toBe("Click me");
});

test("Neutral button Snapshot test", () => {
  let { container } = render(<NeutralButton>Click me</NeutralButton>);
  expect(container).toMatchSnapshot();
  expect(container.textContent).toBe("Click me");
});

test("WarningButton snapshot test", () => {
  let { container } = render(<WarningButton>Click me</WarningButton>);
  expect(container).toMatchSnapshot();
  expect(container.textContent).toBe("Click me");
});

test("Crossbutton snapshot test", () => {
  let { container } = render(<CrossButton>Click me</CrossButton>);
  expect(container).toMatchSnapshot();
  expect(container.textContent).toBe("");
  const burger = within(container).getByTestId("svg-cross");
  expect(burger).toBeDefined();
  expect(burger).toBeVisible();
});
