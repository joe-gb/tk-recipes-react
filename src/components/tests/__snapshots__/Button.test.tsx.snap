// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Crossbutton snapshot test 1`] = `
.c0.c0 {
  z-index: 1;
  margin: 5px;
  background: white;
  border: 1px solid #C9D3DF;
  box-sizing: border-box;
  box-shadow: 0px 1px 2px rgb(0 0 0 / 8%);
  border-radius: 4px;
  padding: 4px;
  height: 32px;
  width: 32px;
}

.c0.c0:hover {
  cursor: pointer;
  border: 1px solid #9EABBE;
  box-shadow: 0px 0px 0px 1px rgb(50 50 93 / 5%),0px 2px 6px rgb(50 50 93 / 8%);
}

.c0.c0 svg {
  fill: #4C5566;
  color: #4C5566;
  vertical-align: middle;
  font-size: 24px;
}

.c0.c0 svg path {
  fill: #4C5566;
}

<div>
  <button
    class="c0"
  >
    <svg
      data-testid="svg-cross"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="M13.414 12l4.293 4.293a1 1 0 01-1.414 1.414L12 13.414l-4.293 4.293a1 1 0 11-1.414-1.414L10.586 12 6.293 7.707a1 1 0 011.414-1.414L12 10.586l4.293-4.293a1 1 0 011.414 1.414L13.414 12z"
        fill-rule="evenodd"
      />
    </svg>
  </button>
</div>
`;

exports[`Neutral button Snapshot test 1`] = `
.c0.c0 {
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 100%;
  margin: 20px auto;
  cursor: pointer;
  font-size: 16px;
  font-weight: 500;
  line-height: 18px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #fff;
  border-radius: 8px;
  background: #003097;
  border: 1px solid #003097;
  padding: 19px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
  -webkit-transition: all 0.175s ease-in-out;
  transition: all 0.175s ease-in-out;
}

.c0.c0:hover {
  background-color: #144fcc;
  border: 1px solid #144fcc;
}

.c1.c1 {
  background: transparent;
  border-color: #9eabbe;
  color: #144fcc;
}

.c1.c1:hover {
  background-color: transparent;
  border: 1px solid #144fcc;
}

<div>
  <button
    class="c0 c1"
  >
    Click me
  </button>
</div>
`;

exports[`Primary button snapshot test 1`] = `
.c0.c0 {
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 100%;
  margin: 20px auto;
  cursor: pointer;
  font-size: 16px;
  font-weight: 500;
  line-height: 18px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #fff;
  border-radius: 8px;
  background: #003097;
  border: 1px solid #003097;
  padding: 19px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
  -webkit-transition: all 0.175s ease-in-out;
  transition: all 0.175s ease-in-out;
}

.c0.c0:hover {
  background-color: #144fcc;
  border: 1px solid #144fcc;
}

<div>
  <button
    class="c0"
  >
    Click me
  </button>
</div>
`;

exports[`WarningButton snapshot test 1`] = `
.c0.c0 {
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 100%;
  margin: 20px auto;
  cursor: pointer;
  font-size: 16px;
  font-weight: 500;
  line-height: 18px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #fff;
  border-radius: 8px;
  background: #003097;
  border: 1px solid #003097;
  padding: 19px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
  -webkit-transition: all 0.175s ease-in-out;
  transition: all 0.175s ease-in-out;
}

.c0.c0:hover {
  background-color: #144fcc;
  border: 1px solid #144fcc;
}

.c1.c1 {
  background: transparent;
  border-color: #003097;
  color: #003097;
}

<div>
  <button
    class="c0 c1"
  >
    Click me
  </button>
</div>
`;
